xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp

```diff
@@ -540,20 +540,28 @@ bool CDVDDemuxFFmpeg::Open(std::shared_ptr<CDVDInputStream> pInput, bool streami
       }
       else
       {
-        // skip programs without or empty audio/video streams
-        for (unsigned int i = 0; nProgram == UINT_MAX && i < m_pFormatContext->nb_programs; i++)
+        // select the program of the first audio or video stream packet found (FIFO)
+        AVStream *st;
+        unsigned int maxpkts = 25;
+        while (nProgram == UINT_MAX && maxpkts-- && (m_pkt.result = av_read_frame(m_pFormatContext, &m_pkt.pkt)) >= 0)
```

Comment:
oh what a hack. you don't accept that that stream from this crappy backend is out of spec. you try to make it work with a hack that harms other streams that comply to the spec. throwing away some packets here ruin the entire GOP.

Reply:
> throwing away some packets here ruin the entire GOP.
Give me a break. Can you help, I'm not a programmer.
It must exist a way to read the packets without throwing away packets, and seek to undo de read.